data/abilities.js

```diff
@@ -735,6 +735,9 @@ exports.BattleAbilities = {
 				return null;
 			}
 		},
+		onEnd: function (pokemon) {
+			pokemon.removeVolatile('flashfire');
```

Comment:
An ability's `onEnd` doesn't remove an ability. When an ability is removed, the battle engine sends an `End` singleEvent to the ability so it can clean up things like this that are linked to the ability. (See Desolate Land's implementation)